src/core/qgsvectorfilewriter.h

```diff
@@ -555,6 +556,25 @@ class CORE_EXPORT QgsVectorFileWriter : public QgsFeatureSink
          * \since QGIS 3.34
          */
         bool includeConstraints = false;
+
+        /**
+         * Set to TRUE to transfer field domains to the exported vector file.
+         *
+         * Support for field domains depends on the output file format.
+         *
+         * \since QGIS 3.36
+         */
+        bool setFieldDomains = true;
+
+        /**
+         * Source database provider connection, for field domains.
+         *
+         * Ownership is not transferred and callers must ensure that the lifetime of srcDatabaseProviderConn
+         * exceeds the lifetime of the QgsVectorFileWriter object.
+         *
+         * \since QGIS 3.36
+         */
+        const QgsAbstractDatabaseProviderConnection *srcDatabaseProviderConn = nullptr;
```

Comment:
We should try to avoid abbreviations in the public APIs, they make the API less "guessable":

```suggestion
        const QgsAbstractDatabaseProviderConnection *sourceDatabaseProviderConnection = nullptr;
```

Reply:
done